search/src/indexer.ts

```diff
@@ -0,0 +1,41 @@
+import MiniSearch from 'minisearch';
+
+const miniSearch = new MiniSearch({
+  fields: ['title', 'headline', 'body', 'section'], // fields to index for full-text search
+  storeFields: ['title', 'headline', 'body', 'section'], // fields to return with search results
+  searchOptions: {
+    boost: { headline: 4, title: 3, section: 2 },
+    fuzzy: 0.2,
+    prefix: true,
+  },
+});
+
+const docBlocks = new Map();
+function indexDoc(content: string, { url, file, title, section, base }) {
+  if (docBlocks.has(file)) miniSearch.removeAll(docBlocks.get(file));
+  let heading: RegExpExecArray;
+  const headingRE = /^\n?(#*)\s+(.*)\n\n([^#]*)/gm;
+  const blocks = [];
+  while ((heading = headingRE.exec(content))) {
+    const [, level, headline, body] = heading;
+    let id = base + url;
+    if (level.length > 1) id += `#${headline.toLowerCase()}`;
```

Comment:
I don't think slug logic should be implemented on the `dockit-core` side
currently slug is compiled in the rehype plugin in Backlight backend when compiling MD files
ideally the logic should come from the same place, but we are duplicating it here and it's out of sync right now (backend replaces also ` ` with `-`, not `20%`)

originally I saw the whole `search` function as a backend thing returned on the `context.search`
but seeing this code I'm starting to think that some things can belong to the client code in `dockit-core`
but not the slug
everything that backend controls should not be duplicated on the client
for slug we might expose `context.slug` or smth to reuse same logic
if you don't like using `context` for this kind of things, let's come up with smth which will contain all functions necessary to be reused between backend and client, I'd gladly search for a better solution

Reply:
to clarify: `context.slug` code should be generated by the backend, but executed on the client, so it's not changing much for the current code of you, but makes a single source of truth about how slug is generated

Reply:
If you read carefully the logic is only here not server side. 
I am using the regexp to extact context vs heading.
It could be done on the server side but it will force a certain API, here I am just exposing the content, it's up to you to extract what you want for the search and replace the full context of search if you want.